### PR TITLE
goresym: init at 1.2

### LINKED ISSUE #174922: packaging request: GoReSym
**Project description**
GoReSym is a Go symbol parser that extracts program metadata (such as CPU architecture, OS, endianness, compiler version, etc), function metadata (start & end addresses, names, sources), filename and line number metadata, and embedded structures and types.

**Metadata**

* homepage URL: https://github.com/mandiant/GoReSym
* source URL: https://github.com/mandiant/GoReSym
* license: MIT
* platforms: Linux

### PR DESCRIPTION
**Pull request by pyrox0**
###### Description of changes
Fixes #174922 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/goresym/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  subPackages = [ "." ];
+
+  vendorSha256 = "sha256-dnQ/tP4RS6WkACobfW7jTTJSHbLrdlZDy1fmO65743Q=";
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Go symbol recovery tool ";
```
Comment (legendofmiracles) [inline pkgs/development/tools/goresym/default.nix:21]:
```suggestion
    description = "Go symbol recovery tool";
```

File: pkgs/development/tools/goresym/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-XDy6qALE2xuOVQHuuvU/gOyTRZ/6CWpyjN3i7VmznoI=";
+  };
+
+  subPackages = [ "." ];
+
+  vendorSha256 = "sha256-dnQ/tP4RS6WkACobfW7jTTJSHbLrdlZDy1fmO65743Q=";
+
+  doCheck = false;
```
Comment (legendofmiracles) [inline pkgs/development/tools/goresym/default.nix:18]: Can you add a comment explaining why the tests must be disabled?

Comment (ModdedGamers) [reply pkgs/development/tools/goresym/default.nix:18]: Fixed. Tests did not need to be disabled.

File: pkgs/development/tools/goresym/default.nix (hunk 1 above)
Comment (ModdedGamers) [reply pkgs/development/tools/goresym/default.nix:21]: Fixed.

**Pull request merged**